### PR TITLE
Mark `Range.detach()` as deprecated + reinstate Firefox support

### PR DESCRIPTION
**Pull request by caugner**
#### Summary

Firefox still has `Range.detach()`, even if it is no-op since Firefox 15 as specified.

#### Test results and supporting details

https://bugzil.la/702948

#### Related issues

Fixes https://github.com/mdn/browser-compat-data/issues/20830.

### CODE REVIEW

File: api/Range.json
Comment (ddbeck) [inline api/Range.json]:
```suggestion
            "deprecated": true
```

**Review by ddbeck — APPROVED** [review]
Arguably the historic implementations (e.g., pre-Firefox 15, pre-Chrome 37, etc.) could be partial implementations of the now-standard behavior. But it's pretty historic so there's not a ton of value in it. It's fine to proceed without that change.

That said, I suggest marking this method as deprecated before merging.

File: api/Range.json
Comment (caugner) [reply api/Range.json]: Agree, should we add this case to https://github.com/mdn/browser-compat-data/blob/main/docs/data-guidelines/README.md#setting-deprecated?
